app/services/service.service.js

```diff
@@ -24,25 +24,29 @@ async function getGatewayAccounts (gatewayAccountIds, correlationId) {
     .map(gatewayAccount => new CardGatewayAccount(gatewayAccount).toMinimalJson())
 }
 
-function updateServiceName (serviceExternalId, serviceName, serviceNameCy, correlationId) {
+async function updateServiceName (serviceExternalId, serviceName, serviceNameCy, correlationId) {
   if (!serviceExternalId) {
     return Promise.reject(new Error(`argument: 'serviceExternalId' cannot be undefined`))
   }
-  return getAdminUsersClient({ correlationId }).updateServiceName(serviceExternalId, serviceName, serviceNameCy)
-    .then(result => {
-      // Update gateway account service names in connector
-      const gatewayAccountIds = lodash.get(result, 'gateway_account_ids', [])
-      return Promise.all(gatewayAccountIds.map(gatewayAccountId => {
+
+  try {
+    const result = await getAdminUsersClient({ correlationId }).updateServiceName(serviceExternalId, serviceName, serviceNameCy)
+
+    const gatewayAccountIds = lodash.get(result, 'gateway_account_ids', [])
+
+    await Promise.all(
+      gatewayAccountIds.map(async gatewayAccountId => {
         if (gatewayAccountId && !isADirectDebitAccount(gatewayAccountId)) {
-          return connectorClient.patchServiceName(gatewayAccountId, serviceName, correlationId)
+          const value = await connectorClient.patchServiceName(gatewayAccountId, serviceName, correlationId)
+          return value
         }
       })
-      )
-        .then(() => result)
-    })
-    .then(result => {
-      return new Service(result)
-    })
+    )
+
+    return new Service(result)
+  } catch (err) {
+    logger.error(`[${correlationId}] Calling connector to update service name failed`)
```

Comment:
we should throw the error here to be handled by the controller

Reply:
I reckon we don't need the logging either - the client code will log an appropriate error. So just remove the try/catch from this function maybe and let any errors propagate through to the controller.

Reply:
Done - removed the try/catch.